packages-content-model/roosterjs-content-model-editor/lib/modelApi/edit/deleteSelection.ts

```diff
@@ -1,31 +1,31 @@
 import { ContentModelDocument } from 'roosterjs-content-model-types';
 import { deleteExpandedSelection } from './utils/deleteExpandedSelection';
+import { FormatWithContentModelContext } from '../../publicTypes/parameter/FormatWithContentModelContext';
 import {
     DeleteResult,
     DeleteSelectionContext,
     DeleteSelectionResult,
     DeleteSelectionStep,
     ValidDeleteSelectionContext,
-    OnDeleteEntity,
 } from './utils/DeleteSelectionStep';
 
 /**
  * @internal
  */
 export function deleteSelection(
     model: ContentModelDocument,
-    onDeleteEntity: OnDeleteEntity,
-    additionalSteps: (DeleteSelectionStep | null)[] = []
+    additionalSteps: (DeleteSelectionStep | null)[] = [],
```

Comment:
I saw that this param was added in some places like the same default value. (mergeModel, ContentModelCopyPastePlugin, insertTable)
What if we move format context before additionalSteps? I think that way we do not need to add the empty array if we do not have additional steps, but still need the format context.

Reply:
I considered about this.

You may noticed that the code under modelApi folder are only operating on an existing Content Model, but have no dependency to editor. That means, I'd like to make all "Model API" are independent to editor, when they are called from editor, we pass in a context object to collect the info that editor needs to know. When they are not called from an editor (that is supported, e.g. a model is created for paste), we don't need to pass any editor related thing. Here this context is something that only editor need to know.

But I think it may be a good idea to reuse EditorContext here, rather than creating a new `FormatWithContentModelContext`. Let me try it.

Reply:
Ok, since the context object also carry in rawEvent which is not directly comes from editor, I think I'd better just keep current change.